src/util/SharedMemory.cpp

```diff
@@ -129,7 +138,29 @@ bool SharedMemory::EnsureCommitted(int size)
     }
     return true;
 #else
-    return false;
+    if( size > max_ )
+        return false;
+    if( size > current_ )
+    {
+        unsigned new_size = current_;
+        while( size > new_size )
+            new_size += windowSize_;
+        void* old_handle = regionHandle;
+        void* new_handle = malloc(new_size);
+        if( new_handle && old_handle )
+        {
+            memcpy(new_handle,old_handle,current_);
+            free(old_handle);
+        }
```

Comment:
Is there a reason to not use `realloc()`?

Reply:
Well, actually it's C++, so if anything I should have used new uint8_t[new_size] and delete[], or even better a std::vector<uint8_t>. But this is only an intermediate solution so that the frontend works and generates IR. The actual challenge will be to transpile the IR.